File: 0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_sum = nums[0]
        max_sum = nums[0]
        n=len(nums)
        for i in range(1, n):
            curr_sum = max(curr_sum+nums[i], nums[i])
            max_sum = max(curr_sum, max_sum)

        return max_sum

File: 0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        // int start=-1;
        // int ans_start=-1,ans_end=-1;
        int curr_sum=0;
        int max_sum=-10000;
        for(int num:nums) {
            // if(curr_sum==0) start=i;
            curr_sum+=num;
            if(curr_sum>max_sum) {
                max_sum=curr_sum;
                // ans_start=start;
                // ans_end=i;
            }
            if(curr_sum<0) curr_sum=0;
        }
        return max_sum;
    }
};
